cyvcf2/tests/test_reader.py

```diff
@@ -1336,3 +1336,31 @@ def test_issue17_no_gt():
     with pytest.raises(Exception):
         for v in vcf:
             v.num_called  # Used to give segmentation fault
+
+
+@pytest.mark.parametrize("path", [
+    "test.vcf.gz",
+    "test-multiallelic-homozygous-alt.vcf.gz",
+    "test-strict-gt-option-flag.vcf.gz",
+    ])
+def test_num_records_indexed(path):
+    vcf = VCF(os.path.join(HERE, path))
+    assert len(list(vcf)) == vcf.num_records
+
+@pytest.mark.parametrize("path", [
+    "test-genotypes.vcf",
+    ])
+def test_num_records_no_index(path):
+    vcf = VCF(os.path.join(HERE, path))
+    with pytest.raises(ValueError, match="must be indexed"):
+        vcf.num_records
+
+def test_num_records_different_index():
+    b = VCF('{}/test.snpeff.bcf'.format(HERE))
+    with pytest.raises(Exception, match="hts_idx_get_stat"):
+        b.num_records
+    assert len(list(b)) == 10
+
+    b.set_index("{}/test-diff.csi".format(HERE))
+    print(b.num_records)
```

Comment:
This fails on the ``hts_idx_get_stat`` call, but it's not clear why. I'm assuming this test case is used to check what happens when you automatically load the wrong index, but then replace with a good one?

Reply:
I'm not sure why it would fail on the first call either!? 
So the print on 1365 works?

Reply:
Yes, I think you could try reindexing, make a new test.snpeff.bcf.csi and see if it still fails

Reply:
This one was pretty weird. It seems that the return value of ``hts_idx_get_stat`` isn't actually diagnostic of errors, and bcftools index -n still reports a value when it is negative. So, I think we just have to ignore it. This ``test.snpeff.bcf`` file works fine when we do that.